### ISSUE TITLE
contact_forces filt

### ISSUE THREAD

**Issue by hulehulema** [yusongmin1/My_unitree_go2_gym#8]
def _reward_trot(self):
        """
        Calculates a reward based on the number of feet contacts aligning with the gait phase. 
        Rewards or penalizes depending on whether the foot contact matches the expected gait phase.
        """
        contact = self.contact_forces[:, self.feet_indices, 2] > 5.

        stance_mask = self._get_gait_phase()
        TROT = (contact[:,0] == contact[:,3]) & \
            (contact[:,1] == contact[:,2]) & \
            (contact[:,0] == stance_mask[:,0]) & \
            (contact[:,1] == stance_mask[:,1])
        # print( JUMP.to(torch.float32).mean())
        # print(self.feet_indices)['FL_foot', 'FR_foot', 'RL_foot', 'RR_foot'] tensor([ 6, 12, 20, 26], device='cuda:0')
        self.trot=TROT.to(torch.float32).mean()*(torch.norm(self.commands[:, :3], dim=1) > 0.1)+(torch.sum((torch.norm(self.contact_forces[:, self.feet_indices, :], dim=-1) > 0.1),dim=1)==4)*(torch.norm(self.commands[:, :3], dim=1) < 0.1)
        # print(stance_mask[0,0],stance_mask[0,1],phase)
        return TROT*(torch.norm(self.commands[:, :3], dim=1) > 0.1)

Is Down Code OK? The contact reporting of PhysX is unreliable on meshes
    def _reward_trot(self):
        """
        Calculates a reward based on the number of feet contacts aligning with the gait phase. 
        Rewards or penalizes depending on whether the foot contact matches the expected gait phase.

        # Need to filter the contacts because the contact reporting of PhysX is unreliable on meshes
        """
        contact = self.contact_forces[:, self.feet_indices, 2] > 5.

        contact = self.contact_forces[:, self.feet_indices, 2] > 1.
       contact_filt = torch.logical_or(contact, self.last_contacts) 
       self.last_contacts = contact

        stance_mask = self._get_gait_phase()
        TROT = (contact_filt [:,0] == contact_filt [:,3]) & \
            (contact_filt [:,1] == contact_filt [:,2]) & \
            (contact_filt [:,0] == stance_mask[:,0]) & \
            (contact_filt [:,1] == stance_mask[:,1])
        # print( JUMP.to(torch.float32).mean())
        # print(self.feet_indices)['FL_foot', 'FR_foot', 'RL_foot', 'RR_foot'] tensor([ 6, 12, 20, 26], device='cuda:0')
        self.trot=TROT.to(torch.float32).mean()*(torch.norm(self.commands[:, :3], dim=1) > 0.1)+(torch.sum((torch.norm(self.contact_forces[:, self.feet_indices, :], dim=-1) > 0.1),dim=1)==4)*(torch.norm(self.commands[:, :3], dim=1) < 0.1)
        # print(stance_mask[0,0],stance_mask[0,1],phase)
        return TROT*(torch.norm(self.commands[:, :3], dim=1) > 0.1)

**Comment by 1874170538** [comment]
这个方法训练，反而效果不好，表象是原地不动，给cmd也一样
    # Need to filter the contacts because the contact reporting of PhysX is unreliable on meshes
    """
    contact = self.contact_forces[:, self.feet_indices, 2] > 5.

    contact = self.contact_forces[:, self.feet_indices, 2] > 1.
   contact_filt = torch.logical_or(contact, self.last_contacts) 
   self.last_contacts = contact

    stance_mask = self._get_gait_phase()
    TROT = (contact_filt [:,0] == contact_filt [:,3]) & \
        (contact_filt [:,1] == contact_filt [:,2]) & \
        (contact_filt [:,0] == stance_mask[:,0]) & \
        (contact_filt [:,1] == stance_mask[:,1])
    # print( JUMP.to(torch.float32).mean())
    # print(self.feet_indices)['FL_foot', 'FR_foot', 'RL_foot', 'RR_foot'] tensor([ 6, 12, 20, 26], device='cuda:0')
    self.trot=TROT.to(torch.float32).mean()*(torch.norm(self.commands[:, :3], dim=1) > 0.1)+(torch.sum((torch.norm(self.contact_forces[:, self.feet_indices, :], dim=-1) > 0.1),dim=1)==4)*(torch.norm(self.commands[:, :3], dim=1) < 0.1)
    # print(stance_mask[0,0],stance_mask[0,1],phase)
    return TROT*(torch.norm(self.commands[:, :3], dim=1) > 0.1)

**Comment by 1874170538** [comment]
原来的给cmd可以正常移动，不过Yaw跟随不好，不是围着身体中心旋转，围着前腿旋转，有什么措施吗？
